bin/gwdetchar-omega

```diff
@@ -433,6 +447,50 @@ html.write_qscan_page(ifo, gps, analyzed, **htmlv)
 # launch omega scans
 gprint('Launching Omega scans...')
 
+# construct matched-filter from primary channel
+if not args.disable_correlation:
+    gprint('Constructing a matched-filter from the primary channel...')
+    duration = primary.duration
+    fftlength = primary.fftlength
+    pad = max(1, fftlength/4.)
+    # check that analysis flag is active for all of `duration`
+    if primary.flag and not args.ignore_state_flags:
+        if args.verbose:
+            gprint('Querying state flag %s...' % primary.flag)
+        active = check_flag(primary.flag, gps, duration, pad)
+    else:
+        active = True
+    # construct a matched-filter
+    if not active:
+        gprint('%s not active, disabling correlation' % primary.flag)
+        args.disable_correlation = True
+    else:
+        # read in `duration` seconds of data centered on gps
+        data = get_data([primary.channel.name], gps, duration, pad,
+                        frametype=primary.frametype, source=primary.source,
+                        nproc=args.nproc, verbose=args.verbose)
+        hoft = data[primary.channel.name].astype('float64')
+        if primary.resample:
+            hoft = hoft.resample(primary.resample)
+        asd = hoft.asd(fftlength, fftlength/2, method='lal_median_mean')
+        corner = primary.channel.frange[0] / 1.5
+        wn = 2 * corner * hoft.dt.decompose().value
+        hpfilt = butter(12, wn, btype='highpass', analog=False, output='sos')
+        hoft = hoft.filter(hpfilt, filtfilt=True)
+        hoft = hoft.whiten(window='hann', asd=asd, detrend='linear')
+        # plot the primary channel
+        if args.verbose:
+            gprint('Plotting a timeseries of the primary channel...')
+        plot.omega_plot(hoft, gps, primary.length, primary.channel.name,
+                        'plots/primary.png', ylabel='Whitened Amplitude',
+                        figsize=[8, 5])
+        hoft = hoft.crop(
+            gps - primary.length/2, gps + primary.length/2).taper()
+        del data
+    # prepare HTML output
+    htmlv['correlated'] = active
+    htmlv['primary'] = primary.channel.name
+
```

Comment:
@alurban, can we move some or all of this block into a function, to make it a bit more modular?

Reply:
@alurban, any thoughts on this?

Reply:
@duncanmmacleod, sorry, I thought I had already responded. Yes, I think that refactoring much of this code block into a function is a very good idea that is on my to-do list. (In fact, much of the [**other refactoring work**](https://github.com/gwpy/gwpy/pull/984) I've been doing has been a preamble to this.)

Reply:
@duncanmmacleod, we discussed this yesterday, and decided it will save effort to get this feature in now and save refactoring the bulk of this code into a function for a future pull request.

Reply:
@duncanmmacleod, I've pushed a commit that _heavily_ refactors `gwdetchar-omega` into a new module, `gwdetchar.omega.core`, that is loaded from `gwdetchar/omega/__init__.py`. It also renames `gwdetchar/omega/scan.py` --> `gwdetchar/omega/matlab.py`, to avoid clobbering a new function I introduced called `scan()` that does all the heavy lifting.